lib/libnxpse050/se050/glue/wraps.c

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ *
+ * OP-TEE implementation for undefined SE050 middleware symbols
+ */
+
+#include <fsl_sss_api.h>
+#include <kernel/delay.h>
+#include <nxLog.h>
+#include <printk.h>
+#include <trace.h>
+#include <wraps.h>
+
+struct strbuf {
+	char buf[MAX_PRINT_SIZE];
+	char *ptr;
+};
+
+#define MIN(a, b)				\
+(__extension__({ __typeof__(a) _a = (a);	\
+__typeof__(b) _b = (b);				\
+_a < _b ? _a : _b; }))
+
+static unsigned long next = 1;
+
+sss_status_t sm_sleep(uint32_t ms)
+{
+	mdelay(ms);
+
+	return kStatus_SSS_Success;
+}
+
+int rand(void)
+{
+	next = next * 1103515245L + 12345;
+
+	return (unsigned int)(next / 65536L) % 32768L;
+}
+
+void srand(unsigned int seed)
+{
+	next = seed;
+}
+
+unsigned int time(void *foo __unused)
+{
+	static int time = 1;
+
+	return time++;
```

Comment:
maybe some generic Core time support.

Reply:
we could that; this function just gets used to generate the seed for the previous rand used in host operations (ie for scp03 mostly); it doesnt really need to be true time of any kind

Reply:
If useless, this could simply return 0.
Otherwise, maybe this needs some atomic access/update.